app/gui/src/dashboard/providers/FeatureFlagsProvider.tsx

```diff
@@ -4,55 +4,82 @@
  * Feature flags provider.
  * Feature flags are used to enable or disable certain features in the application.
  */
-import LocalStorage from '#/utilities/LocalStorage'
-import type { ReactNode } from 'react'
+import { IS_DEV_MODE, isOnElectron } from 'enso-common/src/detect'
 import { z } from 'zod'
 import { createStore, useStore } from 'zustand'
 import { persist } from 'zustand/middleware'
-
-declare module '#/utilities/LocalStorage' {
-  /** Local storage data structure. */
-  interface LocalStorageData {
-    readonly featureFlags: z.infer<typeof FEATURE_FLAGS_SCHEMA>
-  }
-}
-
+import { unsafeWriteValue } from '../utilities/write'
 export const FEATURE_FLAGS_SCHEMA = z.object({
   enableMultitabs: z.boolean(),
   enableAssetsTableBackgroundRefresh: z.boolean(),
   // eslint-disable-next-line @typescript-eslint/no-magic-numbers
   assetsTableBackgroundRefreshInterval: z.number().min(100),
+  enableCloudExecution: z.boolean(),
 })
 
-LocalStorage.registerKey('featureFlags', { schema: FEATURE_FLAGS_SCHEMA })
+/** Feature flags. */
+export type FeatureFlags = z.infer<typeof FEATURE_FLAGS_SCHEMA>
 
 /** Feature flags store. */
-export interface FeatureFlags {
-  readonly featureFlags: {
-    readonly enableMultitabs: boolean
-    readonly enableAssetsTableBackgroundRefresh: boolean
-    readonly assetsTableBackgroundRefreshInterval: number
-  }
-  readonly setFeatureFlags: <Key extends keyof FeatureFlags['featureFlags']>(
+export interface FeatureFlagsStore {
+  readonly featureFlags: FeatureFlags
+  readonly setFeatureFlags: <Key extends keyof FeatureFlags>(
     key: Key,
-    value: FeatureFlags['featureFlags'][Key],
+    value: FeatureFlags[Key],
   ) => void
 }
 
-const flagsStore = createStore<FeatureFlags>()(
+const flagsStore = createStore<FeatureFlagsStore>()(
   persist(
```

Comment:
@somebody1234 I found that the `zustand` has native support for persisting state, wdyt?